mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -467,10 +469,12 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
 
     override def variance(mu: Double): Double = mu * (1.0 - mu)
 
+    private def ylogy(y: Double, mu: Double): Double = {
+      if (y == 0) 0.0 else y * math.log(y / mu)
+    }
+
     override def deviance(y: Double, mu: Double, weight: Double): Double = {
```

Comment:
I would prefer to remove the added method `ylogy`, which is only used in one place, and replace it with the following logic:

````scala
override def deviance(y: Double, mu: Double, weight: Double): Double = {
      val clamped = y.min(1 - epsilon).max(epsilon)
      2.0 * weight * (clamped * math.log(clamped / mu) + (1 - clamped) * math.log((1 - clamped) / (1 - mu)))
    }
````

I think it's better to avoid adding a static method that is only used in one other method, and additionally I think it's clearer what the intent is here. I'm open to other opinions though.